lib/pundit/policy_finder.rb

```diff
@@ -1,4 +1,46 @@
 module Pundit
+  class ScopeResolver
+    attr_reader :policy
+    attr_reader :user
+    attr_reader :resource_scope
+
+    def initialize(user, resource_scope)
+      @user = user
+      @resource_scope = resource_scope.respond_to?(:each) ? resource_scope.last : resource_scope
+      @policy = PolicyFinder.new(resource_scope).policy
+    end
+
+    def resolve
+      scope_class.new(user, resource_scope).resolve if scope_class
+    end
+
+    def resolve!
+      scope_class!.new(user, resource_scope).resolve
+    end
+
+  private
+
+    # @return [nil, Scope{#resolve}] scope class which can resolve to a scope
+    # @see https://github.com/elabs/pundit#scopes
+    # @example
+    #   scope = finder.scope #=> UserPolicy::Scope
+    #   scope.resolve #=> <#ActiveRecord::Relation ...>
```

Comment:
Does this comment still make sense in this new context? Do private methods need doc comments?

Reply:
No it doesn't. Will remove them thanks.